README.md
# iRoller

_Copyright (c) 2012-2013 Daniele Veneroni. Released under MIT License._

iRoller it's an open source web application to roll dice for various game system like:

* d20 System 
* Custom Dice
* Formula Dé
* Catan: The Dice Game
* Ticket to Ride: The Dice Expansion
* Memoir '44
* BattleLore
* C&C: Ancients
* Battles of Westeros

The application is formatted to be easly wrapped on a stand-alone application using PhoneGap Build, so it's available for iOS, Android and Windows Phone 7.

## Supported Platforms:

* **Web App** - [iRoller](http://iroller.venerons.altervista.org)
* **Firefox OS** - [Firefox Marketplace](https://marketplace.firefox.com/app/iroller/)
* **iOS** - _not yet available_
* **Android** - [Play Store](https://play.google.com/store/apps/details?id=org.veneronslabs.iroller), [apk](https://build.phonegap.com/apps/307339/download/android/)
* **Windows Phone** - [xap](https://build.phonegap.com/apps/307339/download/winphone/)


CHANGELOG.md
# iRoller Changelog

_Copyright (c) 2012-2013 Daniele Veneroni. Released under MIT License._

### Update 26/10/2013

* Update to PhoneGap 3.1.0
* Updated android icons
* Updated webapp icons
* Removed WebOS
* Removed BlackBerry
* Removed Symbian

### Update 07/06/2013

* Updated config.xml
* Updated splash screens
* Reordered directories
* Updated copyright

### Update 12/05/2013

* Update to jQuery 2.0.0
* Update to jQuery Mobile 1.3.1
* Update to jQuery UI 1.10.3
* Update to Add2Home 2.0.7
* Code optimization and bug fixs

### Version 3.2.0

* Add roll effect!
* Major performance tweaks
* directory reorder

### Version 3.1.4

* Update jQuery 1.9.1
* Update jQuery Mobile 1.3.0
* Minor fixs

### Version 3.1.3

* Firefox OS-ready
* Fix all CSP Violation
* Add webapp icon 256px

### Version 3.1.2

* Update to PhoneGap 2.2.0
* New splash screens

### Version 3.1.1

* Updated to PhoneGap 2.1.0
* Updated to jQuery 1.8.2
* Updated to jQuery Mobile 1.2.0
* New support for iPhone 5 and iPod Touch 5
* Updated splash screens
* Updated icons for various resolutions

### Version 3.1.0

* New set Battles of Westeros
* New set Ticket to Ride Dice Expansion
* New game system dividers
* New graphics C&C: Ancients
* New show single rolls on Custom Dice
* Improved checks on Custom Dice inputs
* Improved graphics BattleLore
* Code optimizations
* Updated header
* Updated changelog
* Updated readme

### Version 3.0.0

* New XHDPI support
* Updated splash screens
* New interface
* New dice set Catan: The Dice Game
* New dice set Memoir '44
* New dice set BattleLore
* New dice set C&C: Ancients

### Version 2.1.0

* Updated to PhoneGap 2.0.0
* New dice colors in Formula Dé dice simulator

### Version 2.0.0

* New layouts for the results
* History renamed in Logger
* Clear Log now wipe in real time the log

### Version 1.6.0

* Updated to PhoneGap 1.9.0 (lots of bug fix and performance bumps)

### Version 1.5.0

* Updated jQuery Mobile framework to 1.1.1 (lots of bug fix and performance bumps)
* New loading splash screens

### Version 1.4.0

* Updating to the latest framework PhoneGap 1.8.1 (various bug fix)
* New signing certificate

### Version 1.3.0

* Updating to the latest frameworks jQuery 1.7.1 and jQuery Mobile 1.1.0 (This means better performance, better visual effects where supported)


Risorse Extra/iroller-debug.js
/*!
 * iRoller functions
 * Copyright (c) 2012 Daniele Veneroni. Released under MIT License
 */
function init() {
    try {
        localStorage.clear();
        localStorage.logger = "";
    } catch(err) {
        alert("Web Storage is not supported by your browser, some functionality may not work!");
    }
}

function d(n) {
    var oggi = new Date();
    var log = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "] ";
    var ris = "1d" + n + ": ";
    ris += Math.floor(Math.random() * n) + 1;
    log += ris;
    try {
        localStorage.logger = log + "\n" + localStorage.logger;
    } catch(err) {}
    ris = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br /><b><h1>" + ris + "</h1></b>";
    document.getElementById("risultatod20").innerHTML = ris;
}

function coin() {
    var oggi = new Date();
    var log = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "] ";
    var ris = "Coin: ";
    var n = Math.floor(Math.random() * 2);
    if (n == 0) ris += "Tail";
    else ris += "Head";
    log += ris;
    try {
        localStorage.logger = log + "\n" + localStorage.logger;
    } catch(err) {}
    ris = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br /><b><h1>" + ris + "</h1></b>";
    document.getElementById("risultatod20").innerHTML = ris;
}

function custom(num, facce) {
    var oggi = new Date();
    var log = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "] " + num.value + "d" + facce.value + ": ";
    var ris = num.value + "d" + facce.value + ": ";
    var i;
    var somma = 0;
    for (i = 0; i < num.value; i++) {
        somma += Math.floor(Math.random() * facce.value) + 1;
    }
    log += somma;
    ris += somma;
    try {
        localStorage.logger = log + "\n" + localStorage.logger;
    } catch(err) {}
    ris = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br /><b><h1>" + num.value + "d" + facce.value + ": " + somma + "</h1></b>";
    document.getElementById("risultatocustom").innerHTML = ris;
}

function marcia(num) {
    var ris = "";
    var d = 0;
    var oggi = new Date();
    var log = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br />"
    switch (num) {
        case 1:
            d = (Math.floor(Math.random() * 2) + 1);
            ris = log + "<b>1st Gear<br /><h1>" + d + "</h1></b>";
            break;
        case 2:
            d = (Math.floor(Math.random() * 3) + 2);
            ris = log + "<b>2nd Gear<br /><h1>" + d + "</h1></b>";
            break;
        case 3:
            d = (Math.floor(Math.random() * 5) + 4);
            ris = log + "<b>3rd Gear<br /><h1>" + d + "</h1></b>";
            break;
        case 4:
            d = (Math.floor(Math.random() * 6) + 7);
            ris = log + "<b>4th Gear<br /><h1>" + d + "</h1></b>";
            break;
        case 5:
            d = (Math.floor(Math.random() * 10) + 11);
            ris = log + "<b>5th Gear<br /><h1>" + d + "</h1></b>";
            break;
        case 6:
            d = (Math.floor(Math.random() * 10) + 21);
            ris = log + "<b>6th Gear<br /><h1>" + d + "</h1></b>";
            break;
        default:
            alert("Errore nella determinazione della marcia");
    }
    document.getElementById("risultatofd").innerHTML = ris;
    try {
        localStorage.logger = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]     " + num + " Gear:     " + d + "\n" + localStorage.logger;
    } catch(err) {}
}

function catan(num) {
    var d = 0;
    var oggi = new Date();
    document.getElementById("risultatocatan").innerHTML = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br /><br />"
    var i = 0;
    while (i<num) {
        d = (Math.floor(Math.random() * 6) + 1);
        document.getElementById("risultatocatan").innerHTML = document.getElementById("risultatocatan").innerHTML + "<img src=\"lib/images/catan/" + d + ".png\">&nbsp;";
        i++;
    }
}

function memoir44(num) {
    var d = 0;
    var oggi = new Date();
    document.getElementById("risultatomemoir44").innerHTML = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br /><br />"
    var i = 0;
    while (i<num) {
        d = (Math.floor(Math.random() * 6) + 1);
        document.getElementById("risultatomemoir44").innerHTML = document.getElementById("risultatomemoir44").innerHTML + "<img src=\"lib/images/memoir44/" + d + ".png\">&nbsp;";
        i++;
    }
}

function battlelore(num) {
    var d = 0;
    var oggi = new Date();
    document.getElementById("risultatobattlelore").innerHTML = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br /><br />"
    var i = 0;
    while (i<num) {
        d = (Math.floor(Math.random() * 6) + 1);
        document.getElementById("risultatobattlelore").innerHTML = document.getElementById("risultatobattlelore").innerHTML + "<img src=\"lib/images/battlelore/" + d + ".png\">&nbsp;";
        i++;
    }
}

function ancients(num) {
    var d = 0;
    var oggi = new Date();
    document.getElementById("risultatoancients").innerHTML = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br /><br />"
    var i = 0;
    while (i<num) {
        d = (Math.floor(Math.random() * 6) + 1);
        document.getElementById("risultatoancients").innerHTML = document.getElementById("risultatoancients").innerHTML + "<img src=\"lib/images/ancients/" + d + ".png\">&nbsp;";
        i++;
    }
}

function westeros(num) {
    var d = 0;
    var oggi = new Date();
    document.getElementById("risultatowesteros").innerHTML = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br /><br />"
    var i = 0;
    while (i<num) {
        d = (Math.floor(Math.random() * 8) + 1);
        document.getElementById("risultatowesteros").innerHTML = document.getElementById("risultatowesteros").innerHTML + "<img src=\"lib/images/westeros/" + d + ".png\">&nbsp;";
        i++;
    }
}

scripts/iroller.js
// Copyright (c) 2012 Daniele Veneroni. Released under MIT License
"use strict";

// ON LOAD
window.addEventListener("load", function() {
     init();
}, 0);

// FUNZIONE INIT
function init() {
    localStorage.clear();
    localStorage.logger = "";

    var addToHomeConfig = {
        animationIn: 'drop',
        animationOut: 'fade',
        startDelay: 1000,
        lifespan: 5000,
        expire: 2,
        touchIcon:true,
    };

    document.getElementById("d4").onclick = function() { d(4); };
    document.getElementById("d6").onclick = function() { d(6); };
    document.getElementById("d8").onclick = function() { d(8); };
    document.getElementById("d10").onclick = function() { d(10); };
    document.getElementById("d12").onclick = function() { d(12); };
    document.getElementById("d20").onclick = function() { d(20); };
    document.getElementById("d100").onclick = function() { d(100); };
    document.getElementById("coin").onclick = function() { coin(); };

    document.getElementById("custombutton").onclick = function() { custom(); };

    document.getElementById("marcia1").onclick = function() { marcia(1); };
    document.getElementById("marcia2").onclick = function() { marcia(2); };
    document.getElementById("marcia3").onclick = function() { marcia(3); };
    document.getElementById("marcia4").onclick = function() { marcia(4); };
    document.getElementById("marcia5").onclick = function() { marcia(5); };
    document.getElementById("marcia6").onclick = function() { marcia(6); };

    document.getElementById("catan1").onclick = function() { catan(1); };
    document.getElementById("catan2").onclick = function() { catan(2); };
    document.getElementById("catan3").onclick = function() { catan(3); };
    document.getElementById("catan4").onclick = function() { catan(4); };
    document.getElementById("catan5").onclick = function() { catan(5); };
    document.getElementById("catan6").onclick = function() { catan(6); };

    document.getElementById("memoir44-1").onclick = function() { memoir44(1); };
    document.getElementById("memoir44-2").onclick = function() { memoir44(2); };
    document.getElementById("memoir44-3").onclick = function() { memoir44(3); };
    document.getElementById("memoir44-4").onclick = function() { memoir44(4); };
    document.getElementById("memoir44-5").onclick = function() { memoir44(5); };
    document.getElementById("memoir44-6").onclick = function() { memoir44(6); };

    document.getElementById("battlelore1").onclick = function() { battlelore(1); };
    document.getElementById("battlelore2").onclick = function() { battlelore(2); };
    document.getElementById("battlelore3").onclick = function() { battlelore(3); };
    document.getElementById("battlelore4").onclick = function() { battlelore(4); };
    document.getElementById("battlelore5").onclick = function() { battlelore(5); };
    document.getElementById("battlelore6").onclick = function() { battlelore(6); };

    document.getElementById("ancients1").onclick = function() { ancients(1); };
    document.getElementById("ancients2").onclick = function() { ancients(2); };
    document.getElementById("ancients3").onclick = function() { ancients(3); };
    document.getElementById("ancients4").onclick = function() { ancients(4); };
    document.getElementById("ancients5").onclick = function() { ancients(5); };
    document.getElementById("ancients6").onclick = function() { ancients(6); };

    document.getElementById("westeros1").onclick = function() { westeros(1); };
    document.getElementById("westeros2").onclick = function() { westeros(2); };
    document.getElementById("westeros3").onclick = function() { westeros(3); };
    document.getElementById("westeros4").onclick = function() { westeros(4); };
    document.getElementById("westeros5").onclick = function() { westeros(5); };
    document.getElementById("westeros6").onclick = function() { westeros(6); };

    document.getElementById("tickettoridetrain1").onclick = function() { tickettoridetrain(1); };
    document.getElementById("tickettoridetrain2").onclick = function() { tickettoridetrain(2); };
    document.getElementById("tickettoridetrain3").onclick = function() { tickettoridetrain(3); };
    document.getElementById("tickettoridetrain4").onclick = function() { tickettoridetrain(4); };
    document.getElementById("tickettoridetrain5").onclick = function() { tickettoridetrain(5); };

    document.getElementById("tickettoridetunnel").onclick = function() { tickettoridetunnel(3); };

    document.getElementById("refreshlog").onclick = function() { document.getElementById("history").value = localStorage.logger; };
    document.getElementById("clearlog").onclick = function() { localStorage.clear(); localStorage.logger = ""; document.getElementById("history").value = ''; };
    
}

// FUNZIONE D (D20 SYSTEM)
function d(n) {
    var oggi = new Date();
    var log = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]   ";
    var ris = "1d" + n + ":   ";
    ris += Math.floor(Math.random() * n) + 1;
    log += ris;
    localStorage.logger = log + "\n" + localStorage.logger;
    ris = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br /><b><h1>" + ris + "</h1></b>";
    $("#risdiventi").html(ris).effect("bounce", 500);
}

// FUNZIONE COIN (D20 SYSTEM)
function coin() {
    var oggi = new Date();
    var log = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]   ";
    var ris = "Coin: ";
    var n = Math.floor(Math.random() * 2);
    if (n === 0) { ris += "Tail"; }
    else { ris += "Head"; }
    log += ris;
    localStorage.logger = log + "\n" + localStorage.logger;
    ris = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br /><b><h1>" + ris + "</h1></b>";
    $("#risdiventi").html(ris).effect("bounce", 500);
}

// FUNZIONE CUSTOM (CUSTOM DICE)
function custom() {
    var num = document.getElementById("num").value;
    var facce = document.getElementById("facce").value;
    if (num < 1 || num > 100) {
        alert("Number of dice not valid. Min: 1, Max: 100");
        return;
    }
    if (facce < 2 || facce > 100) {
        alert("Number of faces not valid. Min: 2, Max: 100");
        return;
    }
    var oggi = new Date();
    var log = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]   " + num + "d" + facce + ":   ";
    var ris = "[" + oggi.toLocaleDateString() + "&nbsp;" + oggi.toLocaleTimeString() + "]<br /><b><h1>" + num + "d" + facce + "<br /><br />";
    var somma = 0;
    for (var i = 0; i < num; i++) {
        var tiro = Math.floor(Math.random() * facce) + 1;
        somma += tiro;
        ris += "&nbsp;" + tiro + "&nbsp;";
        log += " " + tiro + " ";
    }
    log += "= " + somma;
    localStorage.logger = log + "\n" + localStorage.logger;
    ris += "=&nbsp;" + somma + "</h1></b>";
    $("#risultatocustom").html(ris).effect("bounce", 500);
}

// FUNZIONE MARCIA (FORMULA DE')
function marcia(num) {
    var ris = "",  d, oggi = new Date();
    var log = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br />";
    switch (num) {
        case 1:
            d = (Math.floor(Math.random() * 2) + 1);
            ris = log + "<b>1st Gear<br /><h1>" + d + "</h1></b>";
            break;
        case 2:
            d = (Math.floor(Math.random() * 3) + 2);
            ris = log + "<b>2nd Gear<br /><h1>" + d + "</h1></b>";
            break;
        case 3:
            d = (Math.floor(Math.random() * 5) + 4);
            ris = log + "<b>3rd Gear<br /><h1>" + d + "</h1></b>";
            break;
        case 4:
            d = (Math.floor(Math.random() * 6) + 7);
            ris = log + "<b>4th Gear<br /><h1>" + d + "</h1></b>";
            break;
        case 5:
            d = (Math.floor(Math.random() * 10) + 11);
            ris = log + "<b>5th Gear<br /><h1>" + d + "</h1></b>";
            break;
        case 6:
            d = (Math.floor(Math.random() * 10) + 21);
            ris = log + "<b>6th Gear<br /><h1>" + d + "</h1></b>";
            break;
        default:
            alert("Errore nella determinazione della marcia");
    }
    $("#risultatofd").html(ris).effect("bounce", 500);
    localStorage.logger = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]     " + num + " Gear:     " + d + "\n" + localStorage.logger;
}

// FUNZIONE CATAN (CATAN DICE GAME)
function catan(num) {
    var d, oggi = new Date();
    var ris = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br /><br />";
    var i = 0;
    while (i < num) {
        d = (Math.floor(Math.random() * 6) + 1);
        ris += "<img src=\"scripts/images/catan/" + d + ".png\">&nbsp;";
        i++;
    }
    $("#risultatocatan").html(ris).effect("bounce", 500);
}

// FUNZIONE MEMOIR44 (MEMOIR '44)
function memoir44(num) {
    var d, oggi = new Date();
    var ris = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br /><br />";
    var i = 0;
    while (i < num) {
        d = (Math.floor(Math.random() * 6) + 1);
        ris += "<img src=\"scripts/images/memoir44/" + d + ".png\">&nbsp;";
        i++;
    }
    $("#risultatomemoir44").html(ris).effect("bounce", 500);
}

// FUNZIONE BATTLELORE (BATTLELORE)
function battlelore(num) {
    var d, oggi = new Date();
    var ris = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br /><br />";
    var i = 0;
    while (i < num) {
        d = (Math.floor(Math.random() * 6) + 1);
        ris += "<img src=\"scripts/images/battlelore/" + d + ".png\">&nbsp;";
        i++;
    }
    $("#risultatobattlelore").html(ris).effect("bounce", 500);
}

// FUNZIONE ANCIENTS (C&C: ANCIENTS)
function ancients(num) {
    var d, oggi = new Date();
    var ris = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br /><br />";
    var i = 0;
    while (i < num) {
        d = (Math.floor(Math.random() * 6) + 1);
        ris += "<img src=\"scripts/images/ancients/" + d + ".png\">&nbsp;";
        i++;
    }
    $("#risultatoancients").html(ris).effect("bounce", 500);
}

// FUNZIONE WESTEROS (BATTLES FO WESTEROS)
function westeros(num) {
    var d, oggi = new Date();
    var ris = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br /><br />";
    var i = 0;
    while (i < num) {
        d = (Math.floor(Math.random() * 8) + 1);
        ris += "<img src=\"scripts/images/westeros/" + d + ".png\">&nbsp;";
        i++;
    }
    $("#risultatowesteros").html(ris).effect("bounce", 500);
}

// FUNZIONE TICKETTORIDETRAIN (TICKET TO RIDE DICE GAME)
function tickettoridetrain(num) {
    var d, oggi = new Date();
    var ris = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br /><br />";
    var i = 0;
    while (i < num) {
        d = (Math.floor(Math.random() * 6) + 1);
        ris += "<img src=\"scripts/images/tickettoride/train" + d + ".png\">&nbsp;";
        i++;
    }
    $("#risultatotickettoridetrain").html(ris).effect("bounce", 500);
}

// FUNZIONE TICKETTORIDETUNNEL (TICKET TO RIDE DICE GAME)
function tickettoridetunnel(num) {
    var d, oggi = new Date();
    var ris = "[" + oggi.toLocaleDateString() + " " + oggi.toLocaleTimeString() + "]<br /><br />";
    var i = 0;
    while (i < num) {
        d = (Math.floor(Math.random() * 6) + 1);
        ris += "<img src=\"scripts/images/tickettoride/tunnel" + d + ".png\">&nbsp;";
        i++;
    }
    $("#risultatotickettoridetunnel").html(ris).effect("bounce", 500);
}
